### PR TITLE
Schedule week switching

### PR DESCRIPTION
**Pull request by Thymian18**
# Week switching
This PR adds the possibility to switch between weeks and hence not only show the events of the current week. 

## Problems
In the schedule test file, there are two tests that concern the testing of the arrows that allow to switch between weeks. Since somehow those tests worked fine locally, but not on the CI server, I commented them out. 

Further, there is a test for the back button. Since it redirects to the map activity, the test fails. This has something to do with a google failure and was discovered before when Yann implemented the Dashboard. 

## Preview
image

## Remark
The activity works well, even if there are events that touch multiple weeks. In the future, it might be a good idea to implement caching for the events.

### CODE REVIEW

File: app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt [hunk 1]
```diff
@@ -96,103 +127,173 @@ private class EventDataModifier(val event: ShownEvent) : ParentDataModifier {
...
-    Column(modifier = modifier.testTag(ScheduleActivity.TestTags.SCHEDULE_COLUMN)) {
-        ScheduleHeader(
-            minDate = minDate,
-            maxDate = maxDate,
+    // the starting day is always the monday of the current week
+    val startMonday = LocalDate.now().with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
+
+    var shownWeekMonday by remember { mutableStateOf(startMonday) }
+
+    fun updateCurrentWeekMonday(weeksToAdd: Int) {
```
Comment (Luca-Engel) [inline app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt:161]: Great idea! This keeps us flexible to add features for skipping multiple weeks at a time (e.g., if we decide to add a go-to-current-week button, etc.)

File: app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt [hunk 2]
```diff
@@ -284,26 +366,74 @@ fun BasicSchedule(
...
+            maxLines = 5,
+            overflow = TextOverflow.Ellipsis,
+            fontSize = 10f.sp,
+        )
+    }
+}
+
+
 // --------------------------------------------------
 // mainly for testing, debugging and demo purposes
```
Comment (Luca-Engel) [inline app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt:418]: Once event creation is implemented, this part will be able to be removed.

File: app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt [hunk 3]
```diff
@@ -284,26 +366,74 @@ fun BasicSchedule(
...
 private val currentMonday = LocalDate.now().with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
```
Comment (Luca-Engel) [inline app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt:419]: Once event creation is implemented, this part will be able to be removed.

**Review by Luca-Engel — CHANGES_REQUESTED** [review]
This is a great additional feature and very well implemented! Your tests and code are well designed and very well readable!

I believe, it would be great to add the following test:
An event going from Sunday to Monday and checking if it is displayed on both days. This would test a common edge case as this event affects two different weeks.

Other than that, great job!
